config/src/test/java/org/springframework/security/config/annotation/web/configuration/sec2377/Sec2377Tests.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.config.annotation.web.configuration.sec2377;
+
+import org.junit.Test;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.web.context.support.AnnotationConfigWebApplicationContext;
+
+/**
+ * @author Rob Winch
+ * @author Josh Cummings
+ */
+public class Sec2377Tests {
+
+	@Test
+	public void contextRefreshWhenParentAndChildRegisteredThenNoException() {
+		AnnotationConfigWebApplicationContext parent =
+			new AnnotationConfigWebApplicationContext();
+		parent.register(Sec2377AConfig.class);
+		parent.refresh();
+
+		AnnotationConfigWebApplicationContext child =
+			new AnnotationConfigWebApplicationContext();
+		child.register(Sec2377BConfig.class);
+		child.setParent(parent);
+		child.refresh();
+	}
+
+	@EnableWebSecurity
+	static class Sec2377AConfig extends WebSecurityConfigurerAdapter {
```

Comment:
`Sec2377AConfig` should be in a different package from `Sec2377BConfig`. See the original (groovy) package structure. See original [Issue](https://github.com/spring-projects/spring-security/issues/2598)

Reply:
Okay, no problem. Was the original issue only reproduceable when parent and child configs were in different repos? Otherwise, what is the motive for preserving the package structure?

Reply:
I didn't work on the original issue. Maybe preserving the package structure is not necessary? But the Configs were put in a separate package when this test was originally written so there must be a reason for it?